src/odemis/gui/cont/tabs.py

```diff
@@ -1840,7 +1841,21 @@ def display_new_data(self, filename, data, extend=False):
         all_streams = streams + self.tab_data_model.streams.value
 
         # Spectrum and AR streams are, for now, considered mutually exclusive
+
+        # all spectrum streams including spectrum, temporal spectrum and time correlator (chronograph)
         spec_streams = [s for s in all_streams if isinstance(s, acqstream.SpectrumStream)]
+        self.temporalspectrum_streams = []
+        self.spectrum_streams = []
+        self.timecorrelator_streams = []
+        # sort spec streams
+        for s in spec_streams:
+            if hasattr(s, "selected_wavelength") and hasattr(s, "selected_time"):
+                self.temporalspectrum_streams.append(s)
+            elif hasattr(s, "selected_wavelength") and not hasattr(s, "selected_time"):
+                self.spectrum_streams.append(s)
+            elif not hasattr(s, "selected_wavelength") and hasattr(s, "selected_time"):
+                self.timecorrelator_streams.append(s)
```

Comment:
Currently, selected_wavelength is always created, so all the timecorrelator streams are concidered temporal spectrum, right?
Maybe just add a TODO for now, and later change the SpectrumStaticStream.